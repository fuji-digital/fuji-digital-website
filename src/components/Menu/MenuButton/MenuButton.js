import openMenu from "../../../images/icons/burger-menu.svg";
import closeMenu from "../../../images/icons/cross-icon.svg";

import { StyledMenuButton, StyledMenuImage } from "./MenuButton.styles";

export const MenuButton = ({ isMenuOpen, handleMenuClick }) => {
  return (
    <StyledMenuButton onClick={handleMenuClick}>
      <StyledMenuImage
        src={isMenuOpen ? closeMenu : openMenu}
        alt={isMenuOpen ? "Close menu" : "Open menu"}
        className={isMenuOpen ? "closed" : "open"}
      />
    </StyledMenuButton>
  );
};
