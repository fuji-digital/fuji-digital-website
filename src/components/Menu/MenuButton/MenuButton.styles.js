import styled from "styled-components";

export const StyledMenuButton = styled.button`
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  outline: inherit;
`;

export const StyledMenuImage = styled.img`
  height: 3rem;
  transition: transform 0.3s ease; // Add transition effect

  &.open {
    transform: rotate(0deg);
  }

  &.closed {
    transform: rotate(90deg); // Add a rotate effect for example
  }
`;
