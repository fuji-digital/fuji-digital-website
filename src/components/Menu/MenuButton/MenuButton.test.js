import renderer from "react-test-renderer";
import { render, screen, fireEvent, rerender } from "@testing-library/react";
import { MenuButton } from "./MenuButton";
import closeMenu from "../../../images/icons/cross-icon.svg";
import openMenu from "../../../images/icons/burger-menu.svg";

describe("MenuButton", () => {
  it("on click menu is open", () => {
    let isMenuOpen = false;

    const handleMenuClick = () => {
      isMenuOpen = !isMenuOpen;
    };

    const { rerender } = render(
      <MenuButton isMenuOpen={isMenuOpen} handleMenuClick={handleMenuClick} />
    );
    const menuButton = screen.getByRole(`button`);
    fireEvent.click(menuButton);
    rerender(
      <MenuButton isMenuOpen={isMenuOpen} handleMenuClick={handleMenuClick} />
    );
    const menuImage = screen.getByRole(`img`);

    expect(menuImage).toHaveAttribute("src", closeMenu);

    fireEvent.click(menuButton);
    rerender(
      <MenuButton isMenuOpen={isMenuOpen} handleMenuClick={handleMenuClick} />
    );
    expect(menuImage).toHaveAttribute("src", openMenu);
  });
});
