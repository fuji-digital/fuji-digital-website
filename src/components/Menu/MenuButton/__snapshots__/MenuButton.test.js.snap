// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton matches snapshot 1`] = `
<div
  className="sc-aYaIB VyKKR"
>
  <button
    className="sc-gEvDqW owOns"
    onClick={[Function]}
  >
    <img
      alt="Open menu"
      className="sc-eqUzNf cnDHTq open"
      src="burger-menu.svg"
    />
  </button>
</div>
`;
